augur/util_support/date_disambiguator.py

```diff
@@ -60,7 +60,32 @@ def range(self):
             resolve_uncertain_int(self.uncertain_date_components["m"], "max"),
             resolve_uncertain_int(self.uncertain_date_components["d"], "max"),
         )
-        max_date = min(max_date, datetime.date.today())
+
+        # Set default bounds.
+        lower_bound = None
+        upper_bound = datetime.date.today()
+
+        # TODO: Allow more granularity than year alone.
+        if self.min_max_year:
+            lower_bound = datetime.date(int(self.min_max_year[0]), 1, 1)
+
+            if len(self.min_max_year) == 2:
+                upper_bound = datetime.date(int(self.min_max_year[1]), 12, 31)
+
+        # Limit dates to the given bounds.
+        if lower_bound:
+            if lower_bound > max_date:
```

Comment:
_non-blocking_

It's more intuitive (for me at least) to read this when the comparison is flipped:

```
if max_date < lower_bound:
```

I think this is because I'm thinking of the relative positions of those values on a number line, where the condition is testing if `max_date` is to the left of `lower_bound`.

Reply:
That's an interesting point, it also reads better to me in your suggestion. Probably because the code is read left to right and number scales are typically represented early → late.

I'll assume most others think this way too, so will take the suggestion.

Reply:
However, this seems to read fine (at least for me):

```py
if max_date > upper_bound:
    max_date = upper_bound
```

since `max_date` is more "centric" in the comparison, where the purpose is to impose a limit on its value.

Reply:
Done: 5ab87dbd5186f8ea5125955d8c7a1aaccbf7fccf